### PR TITLE
Spinner + Overlay in v6 via new theme 

### PR DESCRIPTION
**Pull request by akleiner2**
**Overview**
As the title states - moves some stuff around to fit in the new theme styles. Also supports a backwards-compat `size` API. 

**Screenshots (if applicable)**
N/A - purely an underlying CSS composition change. 

**Documentation**

- [ ] Updated Typescript types and/or component PropTypes
- [ ] Added / modified component docs
- [ ] Added / modified Storybook stories

### CODE REVIEW

**Review by colinlohner — APPROVED** [review]
LGTM

File: src/hooks/use-style-config.js [hunk 1]
```diff
@@ -68,6 +68,7 @@ function combineStyles(theme, props, styleConfig, internalStyles = {}) {
     theme,
     props
   )
+
```
Comment (mshwery) [inline src/hooks/use-style-config.js:71]: Hm...

File: src/spinner/src/Spinner.js [hunk 2]
```diff
@@ -72,7 +94,7 @@ const Spinner = memo(
         >
           <Box
             is="circle"
-            className={innerClass(tokens.spinnerColor)}
+            className={innerClass(boxProps.color)}
```
Comment (mshwery) [inline src/spinner/src/Spinner.js:97]: I wonder if we can achieve this via `currentColor`?

File: src/themes/classic/components/spinner.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  color: 'rgba(67, 90, 111, 0.47)'
+}
+const appearances = {}
+const sizes = {
+  small: {
+    width: 32,
+    height: 32
+  },
+  medium: {
+    width: 40,
```
Comment (mshwery) [inline src/themes/classic/components/spinner.js:11]: Spicy 🌶️

**Review by mshwery — APPROVED** [review]

File: src/hooks/use-style-config.js (hunk 1 above)
Comment (akleiner2) [reply src/hooks/use-style-config.js:71]: My b! 😬

Comment (mshwery) [reply src/hooks/use-style-config.js:71]: 😂

**Pull request merged**